algorithm/index.mjs

```diff
@@ -1,17 +1,487 @@
+  });
+
+  const descuentoKs = Math.pow(1 + tasaIndicadorBruto.descuentoKs / 100, inputs.paymentFrequency / 360) - 1;
+  const descuentoWACC = Math.pow(1 + tasaIndicadorNeto.descuentoWACC / 100, inputs.paymentFrequency / 360) - 1;
+
+  const percentageInitialFee = inputs.percentageInitialFee / 100;
+
+  const initialFee = percentageInitialFee * inputs.sellingPrice;
+
+  const newSellingPrice = inputs.sellingPrice - initialFee;
+
+  const loanTimeInDays = inputs.loanTime * 360;
+  const montoIGV = roundMoney((newSellingPrice * IGV) / (1 + IGV));
+  const sellingValue = roundMoney(newSellingPrice - montoIGV);
+
+  const cuotasAnuales = 360 / inputs.paymentFrequency;
+  const periodos = loanTimeInDays / inputs.paymentFrequency;
+
+  let montoRecompra = 0;
+  if (inputs.buyingOption) {
+    montoRecompra = -(sellingValue * (inputs.buyingOptionPercentage / 100));
+  }
+
+  let totalInitialCosts = 0;
+  let periodicCosts = 0;
+  let insuranceAmount = 0;
+
+  for (const extraCost of extraCosts) {
+    if (extraCost.type === "Inicial" && extraCost.valueType === "Monetario") {
+      totalInitialCosts += extraCost.value;
+    } else if (extraCost.type === "Inicial" && extraCost.valueType === "Porcentual") {
+      totalInitialCosts += (extraCost.value / 100) * inputs.sellingPrice;
+    } else if (extraCost.type === "Periódico" && extraCost.valueType === "Monetario") {
+      periodicCosts -= extraCost.value;
+    } else {
+      insuranceAmount = -(((extraCost.value / 100) * newSellingPrice) / cuotasAnuales);
+    }
+  }
+
+  const leasingAmount = totalInitialCosts + sellingValue;
+
+  let periodicalInterestRate = 0;
+
+  if (inputs.interestRateType === "Nominal") {
+    const m = inputs.interestRateFrequency / inputs.capitalizacion;
+    const n = inputs.paymentFrequency / inputs.capitalizacion;
+    periodicalInterestRate = Math.pow(1 + inputs.interestRate / 100 / m, n) - 1;
+  } else {
+    periodicalInterestRate =
+      Math.pow(1 + inputs.interestRate / 100, inputs.paymentFrequency / inputs.interestRateFrequency) - 1;
+  }
+
+  console.log(`\nPrecio de venta del activo: S/ ${inputs.sellingPrice}`);
+  console.log(`Cuota inicial: S/ ${initialFee}\n`);
+  console.log(`Nuevo precio de venta S/ ${newSellingPrice}`);
+  console.log(`Monto de IGV: S/ ${montoIGV}`);
+  console.log(`Valor de venta del activo: S/ ${sellingValue}`);
+  console.log(`Total por costos iniciales: S/ ${totalInitialCosts}`);
+  console.log(`Monto del leasing: S/ ${leasingAmount}`);
+
+  console.log(`\nN° cuotas al año: ${cuotasAnuales}`);
+  console.log(`N° periodos de pago: ${periodos}`);
+  console.log(`Frecuencia de pago: ${inputs.paymentFrequency} días\n`);
+
+  console.log(`Tipo de tasa de interés: ${inputs.interestRateType}`);
+  console.log(`Tasa efectiva del periodo: ${periodicalInterestRate * 100}%\n`);
+
+  console.log(`Tasa descuento Ks equivalente: ${descuentoKs * 100}%`);
+  console.log(`Tasa descuento WACC equivalente: ${descuentoWACC * 100}%`);
```

Comment:
Could you make sure that you use the `Intl.NumberFormat` API also for printing all the results?

Reply:
Done